internal/services/loadbalancer/nat_rule_resource.go

```diff
@@ -307,11 +343,33 @@ func resourceArmLoadBalancerNatRuleDelete(d *pluginsdk.ResourceData, meta interf
 func expandAzureRmLoadBalancerNatRule(d *pluginsdk.ResourceData, lb *network.LoadBalancer, loadBalancerId parse.LoadBalancerId) (*network.InboundNatRule, error) {
 	properties := network.InboundNatRulePropertiesFormat{
 		Protocol:       network.TransportProtocol(d.Get("protocol").(string)),
-		FrontendPort:   utils.Int32(int32(d.Get("frontend_port").(int))),
 		BackendPort:    utils.Int32(int32(d.Get("backend_port").(int))),
 		EnableTCPReset: utils.Bool(d.Get("enable_tcp_reset").(bool)),
 	}
 
+	backendAddressPoolSet, frontendPort := false, false
+	if _, ok := d.GetOk("frontend_port"); ok {
+		frontendPort = true
+	}
+	if _, ok := d.GetOk("backend_address_pool_id"); ok {
+		backendAddressPoolSet = true
+	}
+
+	if backendAddressPoolSet {
+		properties.FrontendPortRangeStart = utils.Int32(int32(d.Get("frontend_port_start").(int)))
+		properties.FrontendPortRangeEnd = utils.Int32(int32(d.Get("frontend_port_end").(int)))
+		properties.BackendAddressPool = &network.SubResource{
+			ID: utils.String(d.Get("backend_address_pool_id").(string)),
+		}
+	} else {
+		if frontendPort {
+			properties.FrontendPort = utils.Int32(int32(d.Get("frontend_port").(int)))
+		} else {
+			properties.FrontendPortRangeStart = utils.Int32(int32(d.Get("frontend_port_start").(int)))
```

Comment:
Since the `backend_address_pool_id`, `frontend_port_start` and `frontend_port_end` are bound together, then shouldn't this branch return error, or?

Reply:
Thanks for pointed this out, checked the api behavior again:
backend_address_pool_id is specified, then the port range properties have to be used, but not the other way around. user can still use the port range properties if they do not specifies the backend_address_pool_id.

thus, changed the logic